hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -375,56 +385,117 @@ public Vertex updateProcessor(
+                builder.dynamicParams(),
+                builder.inputRefs(),
+                table.getBatchLimit()
         );
+
+        return dmlVertex(context, hasInput, table, updatePS, "Update");
     }
 
     @Nonnull
     @Override
-    public Vertex deleteProcessor(@Nonnull DagBuildContext context, @Nullable HazelcastRexNode predicate, boolean hasInput) {
-        // TODO use the predicate
-        assert predicate == null;
+    public Vertex deleteProcessor(
+            @Nonnull DagBuildContext context,
+            @Nullable HazelcastRexNode predicate,
+            boolean hasInput
+    ) {
+        // We don't allow both predicate and input at the same time
+        assert !(predicate != null && hasInput);
+
         JdbcTable table = context.getTable();
 
-        List<String> pkFields = getPrimaryKey(context.getTable())
-                .stream()
-                .map(f -> table.getField(f).externalName())
-                .collect(toList());
+        RexNode rexPredicate = predicate == null ? null : predicate.unwrap(RexNode.class);
 
-        DeleteQueryBuilder builder = new DeleteQueryBuilder(table, pkFields, resolveDialect(table, context));
-        return context.getDag().newUniqueVertex(
-                "Delete(" + table.getExternalNameList() + ")",
-                new DeleteProcessorSupplier(
-                        table.getDataConnectionName(),
-                        builder.query(),
-                        table.getBatchLimit()
+        DeleteQueryBuilder builder = new DeleteQueryBuilder(table,
+                resolveDialect(table, context),
+                rexPredicate,
+                hasInput
+        );
+
+        DMLProcessorSupplier deletePS = new DMLProcessorSupplier(
+                table.getDataConnectionName(),
+                builder.query(),
+                builder.dynamicParams(),
+                builder.inputRefs(),
+                table.getBatchLimit()
+        );
+
+        return dmlVertex(context, hasInput, table, deletePS, "Delete");
+    }
+
+    private static Vertex dmlVertex(
+            DagBuildContext context,
+            boolean hasInput,
+            JdbcTable table,
+            DMLProcessorSupplier processorSupplier,
+            String statement
+    ) {
+
+        if (!hasInput) {
+            // There is no input, we push the whole update query to the database, but we need single fake item
+            // to execute the update in WriteJdbcP
+            // We can consider refactoring the WriteJdbcP, so it doesn't need the fake item in the future.
+
+            // Create random partition key so the fake source and processor run on the same member
+            String partitionKey = UuidUtil.newUnsecureUuidString();
+
+            Vertex v = fakeSourceVertex(context, "FakeSourceFor" + statement, partitionKey);
+            Vertex updateVertex = context.getDag().newUniqueVertex(
+                    statement + "(" + table.getExternalNameList() + ")",
```

Comment:
`statement` text in theory can contain literal secrets and be long. Better name would be `update` + table name + columns or `delete` + table name.

Reply:
ok, statement is only update/delete